spec/index.bs

```diff
@@ -6467,6 +6483,33 @@ interface mixin GPUProgrammablePassEncoder {
             Issue: Check buffer bindings against `minBindingSize` if present.
         </div>
 
+        <div class=validusage>
+            - Let |bufferBindings| be a sequence<{{GPUBufferBinding}} |bufferBinding|>.
+            - Let |textureViews| be an sequence<{{GPUTextureView}} |textureView|>.
+            - For each {{GPUBindGroup}} |bindGroup| in |encoder|.{{GPUProgrammablePassEncoder/[[bind_groups]]}}.
+                - Let {{GPUBindGroupLayoutDescriptor}} |bindGroupLayoutDescriptor| be |bindGroup|.{{GPUBindGroup/[[layout]]}}.{{GPUBindGroupLayout/[[descriptor]]}}.
+                - For each {{GPUBindGroupEntry}} |bindGroupEntry| in |bindGroup|.
+                    - Let {{GPUBindGroupLayoutEntry}} |bindGroupLayoutEntry| be the {{GPUBindGroupLayoutEntry}} inside |bindGroupLayoutDescriptor|.{{GPUBindGroupLayoutDescriptor/entries}} whose {{GPUBindGroupLayoutEntry/binding}} entry is equal to |bindGroupEntry|.{{GPUBindGroupEntry/binding}}.
+                    - If |bindGroupEntry|.{{GPUBindGroupEntry/resource}} is a {{GPUBufferBinding}},
+                        - Let {{GPUBufferBinding}} |bufferBinding| be |bindGroupEntry|.{{GPUBindGroupEntry/resource}}.
+                        - Let {{GPUBufferBindingLayout}} |bufferBindingLayout| be |bindGroupLayoutEntry|.{{GPUBindGroupLayoutEntry/buffer}}. It must not be `null`.
+                        - If |bufferBindingLayout|.{{GPUBufferBindingLayout/type}} is not {{GPUBufferBindingType/"storage"}}, continue.
+                        - For each {{GPUBufferBinding}} |existingBufferBinding| in |bufferBindings|,
+                            - |existingBufferBinding| and |bufferBinding| must not be [=buffer-binding-aliasing=].
+                        - Append |bufferBinding| to |bufferBindings|.
+                    - Otherwise, if |bindGroupEntry|.{{GPUBindGroupEntry/resource}} is a {{GPUTextureView}},
+                        - Let {{GPUTextureView}} |textureView| be |bindGroupEntry|.{{GPUBindGroupEntry/resource}}.
+                        - If |bindGroupLayoutEntry|.{{GPUBindGroupLayoutEntry/storageTexture}} is `null`, continue.
+                        - Let {{GPUStorageTextureBindingLayout}} |storageTextureBindingLayout| be |bindGroupLayoutEntry|.{{GPUBindGroupLayoutEntry/storageTexture}}.
+                        - If |storageTextureBindingLayout|.{{GPUStorageTextureBindingLayout/access}} is not {{GPUStorageTextureAccess/"write-only"}}, continue.
+                        - For each {{GPUTextureView}} |existingTextureView| in |textureViews|,
+                            - |existingTextureView| and |textureView| must not be [=texture-view-aliasing=].
+                        - Append |textureView| to |textureViews|.
+                    - Otherwise, continue.
+
+            Issue(#1842): Revisit this and figure out if we should add a validation error.
```

Comment:
this appears to be the syntax, from checking other usages in this file (sorry I did it wrong in the issue):

```suggestion
            Issue(gpuweb/gpuweb#1842): Revisit this and figure out if we should add a validation error.
```